sky/sdk/lib/widgets/variable_height_scrollable.dart

```diff
@@ -14,18 +12,32 @@ class VariableHeightScrollable extends Scrollable {
   VariableHeightScrollable({
     String key,
     this.builder,
-    this.token
+    this.token,
+    this.layoutState
   }) : super(key: key);
 
   IndexedBuilder builder;
   Object token;
+  BlockViewportLayoutState layoutState;
+
+  // When the token changes the scrollable's contents may have
+  // changed. Remember as much so that after the new contents
+  // have been laid out we can adjust the scrollOffset so that
+  // the last page of content is still visible.
   bool _contentsChanged = true;
 
+  void initState() {
+    assert(layoutState != null);
+    layoutState.addListener(_handleLayoutChanged);
```

Comment:
This still needs a removeListener() somewhere, no? Otherwise each time the layoutState is used with a new VariableHeightScrollable (e.g. in a different part of the app), the previous one will leak.